pkg/cloud/aws/actuators/machine/utils.go

```diff
@@ -138,9 +140,28 @@ func terminateInstances(client awsclient.Client, instances []*ec2.Instance) erro
 
 // ProviderConfigFromMachine gets the machine provider config MachineSetSpec from the
 // specified cluster-api MachineSpec.
-func ProviderConfigFromMachine(machine *clusterv1.Machine) (*providerconfigv1.AWSMachineProviderConfig, error) {
+func ProviderConfigFromMachine(client client.Client, machine *clusterv1.Machine) (*providerconfigv1.AWSMachineProviderConfig, error) {
+	var providerConfig runtime.RawExtension
+
+	// If no machine.Spec.ProviderConfig.Value then get lookup machineClass
+	if machine.Spec.ProviderConfig.Value != nil {
+		providerConfig = *machine.Spec.ProviderConfig.Value
+	} else {
+		machineClass := &clusterv1.MachineClass{}
+		if machine.Spec.ProviderConfig.ValueFrom != nil && machine.Spec.ProviderConfig.ValueFrom.MachineClass != nil {
+			key := types.NamespacedName{
+				Namespace: machine.Spec.ProviderConfig.ValueFrom.MachineClass.Namespace,
+				Name:      machine.Spec.ProviderConfig.ValueFrom.MachineClass.Name,
+			}
+			if err := client.Get(context.Background(), key, machineClass); err != nil {
+				return nil, err
+			}
+			providerConfig = machineClass.ProviderConfig
+		}
```

Comment:
Missing

```go
} else {
	return fmt.Errorf("unable to find machine provider config: neither Spec.ProviderConfig.Value nor Spec.ProviderConfig.ValueFrom set"), nil
}
```

Reply:
Presumably the return arguments are they other way around here.

Reply:
Yeah, you are right. Though, @enxebre implemented that case differently.